### PR TITLE
Added Coursera Search Engine

### PR DESCRIPTION
**Pull request by sp1thas**
---
name: Engine Implementation
about: Implemented a new engine
title: 'Coursera Search Engine'
labels: 'engine', 'needs-review'
assignees: '@deven96', '@MenSaaH'

---

**Summary of steps Taken to implement the engine**
This is a search engine for coursera.org

This PR resolves issue #29 
```
Ex. I added the Coursera engine, `coursera.py` to the `search_engine_parser/core/engines` directory and made the necessary imports.
This engine integrates Coursera search capabilities and returns titles, links, difficulties, ratings, partners, enrolments etc.

### CODE REVIEW

**Comment by deven96** [comment]
Please make the commit to master, fix the conflicts and squash your commits into ome

File: search_engine_parser/core/engines/coursera.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+from search_engine_parser.core.base import BaseSearch
+from urllib.parse import urljoin
+
+
+class CourseraSearch(BaseSearch):
+    """
+    Searches Coursera for string
+    """
+    name = "Coursera"
+    search_url = "https://www.coursera.org/search?query={query}&indices[prod_all_products_term_optimization_test][page]={page}"
```
Comment (deven96) [inline search_engine_parser/core/engines/coursera.py]: Create a `base_url` so as to be able to use it later on

File: search_engine_parser/core/engines/coursera.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    summary = "\tCoursera is an American online learning platform founded by Stanford professors Andrew Ng and " \
+              "Daphne Koller that offers massive open online courses, specializations, and degrees."
+
+    def parse_soup(self, soup):
+        """
+        Parses Coursera Search Soup for results
+        """
+        # find all class_='gs_r gs_or gs_scl' => each result
```
Comment (deven96) [inline search_engine_parser/core/engines/coursera.py]: Google scholar comment

File: search_engine_parser/core/engines/coursera.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        return soup.find_all('li', class_='ais-InfiniteHits-item')
+
+    def parse_single_result(self, single_result):
+        """
+        Parses the source code to return
+
+        :param single_result: single result found in <div class="gs_r gs_or gs_scl">
```
Comment (deven96) [inline search_engine_parser/core/engines/coursera.py]: Google scholar comment

File: docs/supported_engines.md [hunk 4]
```diff
@@ -21,3 +21,4 @@ Below is the list of supported engines with their summaries
 11|YouTube|titles, links, descriptions, channels, [single videos only: durations, views, upload_dates]
 12|MyAnimeList|titles, links, descriptions, number of episodes, type of result (OVA, series, movie, etc.), ratings
 13|GoogleScholar|titles, links, descriptions, type of results ([BOOK], [CITATION], etc.), links of files
+14|Coursera|titles,links,ratings count, ratings average, partners, difficulties, enrolments numbers
```
Comment (PalAditya) [inline docs/supported_engines.md]: Just an observation, you should probably add spaces after the commas (initial three) to make it look consistent 😅

**Pull request merged**